app/src/main/java/com/intervaltimer/google/activities/MainActivity.java

```diff
@@ -142,51 +141,46 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
         }
 
         Intent receivedIntent = getIntent();
-        Log.i(LONG_INTENT_TAG, "Intent in onCreate() triggered");
+        Log.i(LOG_INTENT_TAG, "Intent in onCreate() triggered");
         handleIntent(receivedIntent);
     }
 
     @Override
     protected void onNewIntent(final Intent intent) {
         super.onNewIntent(intent);
-        Log.i(LONG_INTENT_TAG, "Intent triggered");
+        Log.i(LOG_INTENT_TAG, "Intent triggered");
         handleIntent(intent);
     }
 
     public void handleIntent(final Intent receivedIntent) {
         // EXERCISE_START
-        Toast show;
+
         if (receivedIntent != null) {
```

Comment:
What about reducing the overall nesting with the following refactoring

if (receivedIntent == null) {
   return;
}

// remaining code



Same strategy can be applied below